src/qibo/tests/test_derivative.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import pytest
+
+from qibo import gates, hamiltonians
+from qibo.derivative import parameter_shift
+from qibo.models import Circuit
+
+
+# defining an observable
+def hamiltonian(nqubits=1):
+    m0 = (1 / nqubits) * hamiltonians.Z(nqubits).matrix
+    ham = hamiltonians.Hamiltonian(nqubits, m0)
+    return ham
```

Comment:
This also needs to be adjusted for the backend
```suggestion
def hamiltonian(nqubits, backend):
    m0 = (1 / nqubits) * hamiltonians.Z(nqubits, backend=backend).matrix
    ham = hamiltonians.Hamiltonian(nqubits, m0, backend=backend)
    return ham
```
Also why you do not return
```py
return (1 / nqubits) * hamiltonians.Z(nqubits, backend=backend)
```
directly? I believe it will have exactly the same effect.